### ISSUE TITLE
test_types_in_equals false positive

### ISSUE THREAD

**Issue by alexeieleusis** [dart-lang/linter#443]
```dart
bool operator ==(Object other) {
    if (other is! Foo) return false;
   ...
}
```

**Comment by srawlins** [comment]
To clarify the example:

```dart
class Foo {
  bool b = false;
  @override
  bool operator ==(Object other) {
    if (other is! Foo) return false;
    Foo otherFoo = other as Foo;
    return otherFoo.b;
  }

  @override
  int get hashCode => 7;
}
```

results in a lint report on line 6.

**Comment by asashour** [comment]
Should be closed, as discussed in #2834. With `null-safety`, the issue is not relevant.

**Comment by pq** [comment]
Agreed.  Thanks!
